client/js/libs/handlebars/ircmessageparser/findLinks.js

```diff
@@ -0,0 +1,63 @@
+"use strict";
+
+const URI = require("urijs");
+
+// Known schemes to detect in a text. If a text contains `foo...bar://foo.com`,
+// the parsed scheme should be `foo...bar` but if it contains
+// `foo...http://foo.com`, we assume the scheme to extract will be `http`.
+const commonSchemes = [
+	"http", "https",
+	"ftp", "sftp",
+	"smb", "file",
+	"irc", "ircs",
+	"svn", "git",
+	"steam", "mumble", "ts3server",
+	"svn+ssh", "ssh",
+];
+
+function findLinks(text) {
+	let result = [];
+	let lastPosition = 0;
+
+	// URI.withinString() identifies URIs within text, e.g. to translate them to
+	// <a>-Tags.
+	// See https://medialize.github.io/URI.js/docs.html#static-withinString
+	// In our case, we store each URI encountered in a result array.
+	URI.withinString(text, function(url, start, end) {
+		// v-- fix: url was modified and does not match input string -> cant be mapped
+		if (text.indexOf(url, lastPosition) < 0) {
+			return;
+		}
+		// ^-- /fix: url was modified and does not match input string -> cant be mapped
```

Comment:
@Bonuspunkt, @xPaw, I'm still confused: is this fix still relevant?
There is no test case for it (i.e. coverage does not go inside that `if` and commenting it out does not make the test suite to fail). If it's still relevant, what would be a good example to add to the test suite? If it's not relevant, then we should remove it.